### PR TITLE
Add compose example

### PR DESCRIPTION
**Pull request by jpm-canonical**
Move docker companion snap into subdirectory, and add another example using docker compose.

### CODE REVIEW

File: companion-snap-compose/README.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    env.tc-uri=wss://eu1.cloud.thethings.network:8887 \
+    env.device=/dev/spidev0.0 \
+    env.interface=SPI \
+    env.spi-speed=200000 \
+    env.use-libgpiod=1 \
+    env.gpio-chip=gpiochip4 \
+    env.reset-gpio=22
+```
+
+An option can be cleared with the `snap unset` command, ex:
```
Comment (farshidtz) [inline companion-snap-compose/README.md:65]:
```suggestion
An option can be cleared with the `snap unset` command, e.g.:
```

File: companion-snap-compose/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    command: start.sh
+    install-mode: disable
+    restart-delay: 60s
+
+  gateway-eui:
+    plugs: [docker, docker-executables]
+    command: run-in-container.sh gateway_eui
+
+  find-concentrator:
+    plugs: [docker, docker-executables]
```
Comment (farshidtz) [inline companion-snap-compose/snap/snapcraft.yaml:72]: Maybe reduce the 5 instances of `[docker, docker-executables]` by setting an anchor at first and using aliases for the remaining.

File: companion-snap-compose/start.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -eu
+
+echo "Running: docker compose -f $SNAP/docker-compose.yaml up"
+docker compose -f $SNAP/docker-compose.yaml up
+# docker compose up
```
Comment (farshidtz) [inline companion-snap-compose/start.sh:5]: Code in comments is redundant and hard to maintain. Turn on xtrace instead?

```suggestion
#!/bin/bash -eux

docker compose -f $SNAP/docker-compose.yaml up
```

File: companion-snap-compose/README.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+You can also verify the options with `snap get`.
+For example directly after installation you will see a number of defaults that are already set:
+
+```
+$ sudo snap get lora-basicstation env
+Key            Value
+env.device     /dev/spidev0.0
+env.gpio-chip  gpiochip0
+```
```
Comment (farshidtz) [inline companion-snap-compose/README.md:79]: Where are these defaults set? Not in the install hook.

File: companion-snap-compose/README.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+This snap has the `env.device` and `env.gpio-chip` configs set by default.
```
Comment (farshidtz) [inline companion-snap-compose/README.md:81]:
```suggestion
```

File: companion-snap-compose/run-in-container.sh
Comment (farshidtz) [inline companion-snap-compose/run-in-container.sh:1]: This was hard to follow without looking at the snapcraft file and this script. Rename to run-command-in-container.sh?

**Review by farshidtz — COMMENTED** [review]
I can't fully test this as I don't have the hardware. I've done a code review on the compose additions. It looks good overall.

File: companion-snap-compose/start.sh (hunk 3 above)
Comment (jpm-canonical) [reply companion-snap-compose/start.sh:5]: I'm just removing the echo, as it does not really provide any benefits.

File: companion-snap-compose/README.md (hunk 4 above)
Comment (jpm-canonical) [reply companion-snap-compose/README.md:79]: It got dropped in the compose snap. Adding it back in to the install hook.

**Review by farshidtz — APPROVED** [review]
